src/odemis/gui/cont/acquisition.py

```diff
@@ -450,6 +450,455 @@ def open_acquisition_dialog(self):
+            if s not in item_stream:
+                self._panel.streams_chk_list.Insert(s.name.value, i, s)
+                self._panel.streams_chk_list.Check(i)
+                self._acquiStreams.value.append(s)
+                logging.info("A stream was added")
+                # subscribe the name and wavelength VA's of this added stream
+                for va in self._get_name_wavelength_vas(s):
+                    va.subscribe(self._on_name_and_wavelength, init=False)
+                # subscribe the settings VA's of this added stream
+                for va in self._get_settings_vas(s):
+                    va.subscribe(self._on_settings_va)
+                self._update_acquisition_time()
+
+    @wxlimit_invocation(1)  # max 1/s
+    def _update_acquisition_time(self):
+        """
+        Updates the estimated time
+        required for acquisition
+        """
+        acq_time = acqmng.estimateTime(self._acquiStreams.value)
+        acq_time = math.ceil(acq_time)  # round a bit pessimistic
+        # TODO take into account the time of acquiring the zlevels
+        # as well as the time required for moving the actuator
+        # if self._zStackActive.value:
+        #     acq_time = acq_time * len(generate_zlevels(....))
+        txt = u"Estimated time: {}."
+        txt = txt.format(units.readable_time(acq_time, full=False))
+        self._panel.txt_cryosecom_est_time.SetLabel(txt)
+
+    @call_in_wx_main
+    def _on_name_and_wavelength(self, VA_value):
+        """
+        a VA callback that is called when the user changes
+         the name of a stream or the emission/excitation wavelength
+        """
+        if isinstance(VA_value, str):  # the user changed the stream name
+            counts = self._panel.streams_chk_list.GetCount()
+            chcklist_names = [
+                self._panel.streams_chk_list.GetString(i) for i in range(counts)
+            ]
+            streams = [
+                self._panel.streams_chk_list.GetClientData(i) for i in range(counts)
+            ]
+            stream_names = [s.name.value for s in streams]
+            for name, i, s in zip(chcklist_names, range(counts), streams):
+                if name not in stream_names:
+                    self._panel.streams_chk_list.Delete(i)
+                    self._panel.streams_chk_list.Insert(VA_value, i, s)
+                    self._panel.streams_chk_list.Check(i)
+        else:  # the user changed the wavelength
+            counts = self._panel.streams_chk_list.GetCount()
+            streams = [
+                self._panel.streams_chk_list.GetClientData(i) for i in range(counts)
+            ]
+            for i, s in zip(range(counts), acqmng.sortStreams(streams)):
+                # replace the unsorted streams with the sorted streams one by one
+                self._panel.streams_chk_list.Delete(i)
+                self._panel.streams_chk_list.Insert(s.name.value, i, s)
+                self._panel.streams_chk_list.Check(i)
```

Comment:
i thought because changing the stream name leads to changing the wavelengths, I thought it's more compact to combine the similar parts of code in one function.